spam/03_spam_data_slicing_tutorial.py

```diff
@@ -362,80 +249,198 @@ def textblob_polarity(x):
 )
 
 # %% [markdown]
-# ## 5. Improve slice performance
+# ## 3. Improve slice performance
 #
 # In classification tasks, we might attempt to increase slice performance with techniques like _oversampling_ (i.e. with PyTorch's [`WeightedRandomSampler`](https://pytorch.org/docs/stable/data.html#torch.utils.data.WeightedRandomSampler)).
 # This would shift the training distribution to over-represent certain minority populations.
-# Intuitively, we'd like to show the model more `short_link` examples so that the representation is better suited to handle them.
+# Intuitively, we'd like to show the model more `short_link` examples so that *the representation is better suited to handle examples in this slice*.
 #
 # A technique like upsampling might work with a small number of slices, but with hundreds or thousands or production slices, it could quickly become intractable to tune upsampling weights per slice.
 # In the following section, we show a modeling approach that we call _Slice-based Learning,_ which handles numerous slices using with slice-specific representation learning.
 
+# %% [markdown]
+# ### Set up modeling pipeline with `BinarySlicingClassifier`
```

Comment:
Why this name?  Can we just make it `SlicingClassifier`

Reply:
To emphasize the limited support. But we can state this in the docstrings for now.